### PR TITLE
impl guide: Update Collator Generation

### PR DESCRIPTION
**Pull request by mrcnski**
Re-raising changes I made in https://github.com/paritytech/polkadot/pull/6938, with async-backing content removed.

I think I added the "Summary in plain English" section because we are trying to move away from code in the impl guide, but let me know if it's not necessary.

### CODE REVIEW

File: roadmap/implementers-guide/src/node/collators/collation-generation.md [hunk 1]
```diff
@@ -94,15 +109,85 @@ pub struct CollationGenerationConfig {
...
+      chain.
+
+- **Collation generation config**
+
+  - Contains collator's authentication key, collator function, and
+    parachain ID.
+
+## With Async Backing
+
+### Protocol
```
Comment (slumber) [inline roadmap/implementers-guide/src/node/collators/collation-generation.md:144]: Was it discussed somewhere? As long as we don't have a plan for collation generation, we shouldn't put it into the guide. We're still in the middle of cumulus changes.

Comment (mrcnski) [reply roadmap/implementers-guide/src/node/collators/collation-generation.md:144]: Oops sorry, thought I removed this section.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2863168

**Review by BradleyOlson64 — APPROVED** [review]

File: roadmap/implementers-guide/src/node/collators/collation-generation.md [hunk 2]
```diff
@@ -94,15 +109,42 @@ pub struct CollationGenerationConfig {
...
+- **Collation generation config**
+
+  - Contains collator's authentication key, collator function, and
+    parachain ID.
+
+## Glossary
+
+- *Slot:* Time is divided into discrete slots. Each validator in the validator
+  set produces a verifiable random value, using a VRF, per slot. If below a
+  threshold, this allows the validator to author a new block for that slot.
```
Comment (slumber) [inline roadmap/implementers-guide/src/node/collators/collation-generation.md:146]: leftover

**Review by slumber — APPROVED** [review]

**Comment by mrcnski** [comment]
bot merge
